refinery/core/api.py

```diff
@@ -677,9 +677,15 @@ def filter_by_group(self, request, obj_list):
                 group = None
 
             if group:
-                obj_list = list(get_objects_for_group(
+                filtered_obj_list = []
+                group_obj_list = list(get_objects_for_group(
```

Comment:
It seems odd that `get_objects_for_group` is returning more than what we want, but if thats the case you could save a few lines with an approach like:
```
objects_for_group = get_objects_for_group(
    group, 'core.read_meta_dataset'
)
# get_objects_for_group returns too much, filter against initial obj_list
return [obj for obj in objects_for_group if obj in obj_list]
```

Also maybe making the comment more clear could be helpful. I'm not sure what the `ref in bundle` portion means in this case

Reply:
@scottx611x Take a look at get_object_list
filter_by_group is a helper method, 'referenced in data bundle'. I can update the comment.

We're grabbing the (owned/public) filtered obj list and then grabbing the group filtered list. I just added the intersection code. Perviously, the code just dropped the first filtered obj list if a group filter was added.

I think it's worth a few lines of extra code if it's easier to read.